addon/resolver.js

```diff
@@ -273,34 +280,53 @@ var Resolver = DefaultResolver.extend({
     }
   },
 
-  chooseModuleName: function(moduleName) {
+  chooseModuleName: function(moduleName, parsedName) {
     var underscoredModuleName = underscore(moduleName);
+    let result;
 
     if (moduleName !== underscoredModuleName && this._moduleRegistry.has(moduleName) && this._moduleRegistry.has(underscoredModuleName)) {
       throw new TypeError("Ambiguous module names: `" + moduleName + "` and `" + underscoredModuleName + "`");
     }
 
     if (this._moduleRegistry.has(moduleName)) {
-      return moduleName;
+      result = moduleName;
     } else if (this._moduleRegistry.has(underscoredModuleName)) {
-      return underscoredModuleName;
-    } else {
+      result = underscoredModuleName;
+    }
+    Ember.runInDebug(() => {
```

Comment:
Since the `partializedModuleName` stuff below actually changes the return value, I think it does need to run in both debug and production builds. 

Can you move the `runInDebug` down to only wrap the `isCamelCaseHelper` stuff below?